ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/style/ImageComponentStyle.kt

```diff
@@ -0,0 +1,90 @@
+package com.revenuecat.purchases.ui.revenuecatui.components.style
+
+import androidx.compose.foundation.isSystemInDarkTheme
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.Immutable
+import androidx.compose.runtime.ReadOnlyComposable
+import androidx.compose.ui.graphics.Shape
+import androidx.compose.ui.layout.ContentScale
+import com.revenuecat.purchases.paywalls.components.properties.ColorInfo
+import com.revenuecat.purchases.paywalls.components.properties.Size
+import java.net.URL
+
+@Suppress("LongParameterList")
+@Immutable
+internal class ImageComponentStyle private constructor(
+    @get:JvmSynthetic
+    val visible: Boolean,
+    @get:JvmSynthetic
+    val url: URL,
+    @get:JvmSynthetic
+    val lowResURL: URL?,
+    @get:JvmSynthetic
+    val darkURL: URL?,
+    @get:JvmSynthetic
+    val darkLowResURL: URL?,
+    @get:JvmSynthetic
+    val size: Size,
+    @get:JvmSynthetic
+    val shape: Shape?,
+    @get:JvmSynthetic
+    val gradientColors: ColorInfo.Gradient?,
+    @get:JvmSynthetic
+    val darkGradientColors: ColorInfo.Gradient?,
+    @get:JvmSynthetic
+    val contentScale: ContentScale,
```

Comment:
Note that this might require some transformations from the `ImageComponent` to this `ImageComponentStyle`. For example, this expects a `ContentScale` but in `ImageComponent` we have our own `FitMode`. I believe this translation should be done separately and in advance though.